handlers/generic.go

```diff
@@ -12,179 +12,97 @@
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
 
-package parser
+package handlers
 
 import (
-	"archive/tar"
 	"bytes"
-	"compress/gzip"
-	"crypto/sha256"
-	"encoding/hex"
 	"io"
 	"io/ioutil"
+	"os"
 	"path/filepath"
 	"strings"
 
-	"github.com/mendersoftware/mender-artifact/metadata"
+	"github.com/mendersoftware/mender-artifact/artifact"
 	"github.com/pkg/errors"
 )
 
-type GenericParser struct {
-	metadata   metadata.Metadata
-	updates    map[string]UpdateFile
-	typeParsed string
+type Generic struct {
```

Comment:
Generic what? I think it should have "Update" in the name. That's what it is, right?